scripts/gitlab/check_runtime.sh

```diff
@@ -52,176 +52,178 @@ runtime_regex="^runtime/$(join_by '|^runtime/' "${runtimes[@]}" "${common_dirs[@
 
-		if git diff --name-only "origin/master...${CI_COMMIT_SHA}" \
-			| grep -E -q -e "$regex"
-		then
-			# add label breaksapi only if this pr altered the runtime sources
-			github_label "D2-breaksapi"
-		fi
+  # see if the version and the binary blob changed
+  if [ "${add_spec_version}" != "${sub_spec_version}" ]
+  then
 
-		boldcat <<-EOT
-			## RUNTIME: ${RUNTIME} ##
+    if git diff --name-only "origin/master...${CI_COMMIT_SHA}" \
+      | grep -E -q -e "$regex"
+    then
+      # add label breaksapi only if this pr altered the runtime sources
+      github_label "D2-breaksapi"
+    fi
 
-			changes to the ${RUNTIME} runtime sources and changes in the spec version.
+    boldcat <<EOT
+## RUNTIME: ${RUNTIME} ##
 
-			spec_version: ${sub_spec_version} -> ${add_spec_version}
+changes to the ${RUNTIME} runtime sources and changes in the spec version.
 
-		EOT
-		continue
+spec_version: ${sub_spec_version} -> ${add_spec_version}
 
-	else
-		# check for impl_version updates: if only the impl versions changed, we assume
-		# there is no consensus-critical logic that has changed.
+EOT
+    continue
 
-		add_impl_version="$(
-			git diff refs/tags/"${LATEST_TAG}...${CI_COMMIT_SHA}" "runtime/${RUNTIME}/src/lib.rs" \
-			| sed -n -r 's/^\+[[:space:]]+impl_version: +([0-9]+),$/\1/p'
-		)"
-		sub_impl_version="$(
-			git diff refs/tags/"${LATEST_TAG}...${CI_COMMIT_SHA}" "runtime/${RUNTIME}/src/lib.rs" \
-			| sed -n -r 's/^\-[[:space:]]+impl_version: +([0-9]+),$/\1/p'
-		)"
+  else
+    # check for impl_version updates: if only the impl versions changed, we assume
+    # there is no consensus-critical logic that has changed.
 
+    add_impl_version="$(
+      git diff refs/tags/"${LATEST_TAG}...${CI_COMMIT_SHA}" "runtime/${RUNTIME}/src/lib.rs" \
+      | sed -n -r 's/^\+[[:space:]]+impl_version: +([0-9]+),$/\1/p'
+    )"
+    sub_impl_version="$(
+      git diff refs/tags/"${LATEST_TAG}...${CI_COMMIT_SHA}" "runtime/${RUNTIME}/src/lib.rs" \
+      | sed -n -r 's/^\-[[:space:]]+impl_version: +([0-9]+),$/\1/p'
+    )"
 
-		# see if the impl version changed
-		if [ "${add_impl_version}" != "${sub_impl_version}" ]
-		then
-			boldcat <<-EOT
 
-			## RUNTIME: ${RUNTIME} ##
+    # see if the impl version changed
+    if [ "${add_impl_version}" != "${sub_impl_version}" ]
+    then
+      boldcat <<EOT
 
-			changes to the ${RUNTIME} runtime sources and changes in the impl version.
+## RUNTIME: ${RUNTIME} ##
 
-			impl_version: ${sub_impl_version} -> ${add_impl_version}
+changes to the ${RUNTIME} runtime sources and changes in the impl version.
 
-			EOT
-			continue
-		fi
+impl_version: ${sub_impl_version} -> ${add_impl_version}
 
+EOT
+      continue
+    fi
 
-		boldcat <<-EOT
-		wasm source files changed or the spec version in the substrate reference in
-		the Cargo.lock but not the spec/impl version. If changes made do not alter
-		logic, just bump 'impl_version'. If they do change logic, bump
-		'spec_version'.
+    failed_runtime_checks+=($RUNTIME)
```

Comment:
```suggestion
    failed_runtime_checks+=("$RUNTIME")
```

Fixing the shellcheck nag. But I like this idea! Much more useful output from the job if it fails.

Reply:
Sorry for that! I just installed shellcheck and will make sure all warnings are gone